Source/Core/Core/AchievementManager.h

```diff
@@ -152,6 +155,7 @@ class AchievementManager
   Core::System* m_system{};
   bool m_is_runtime_initialized = false;
   UpdateCallback m_update_callback;
+  const DiscIO::Volume* m_loading_volume;
```

Comment:
Having this as a class member is very questionable, since you don't have any guarantees about the lifetime of the volume except that the volume will exist for the duration of the LoadGameAsync call.

This variable is only used in LoadGameAsync, so I would recommend making it a local variable in LoadGameAsync instead. You can capture it in the lambda.

Reply:
I'm not sure I _can_ capture it in the lambda, it needs to parse that as a C function pointer and I haven't had any luck finding a way to do that yet.

Reply:
Ah. Yeah, I see the problem then.

Reply:
I'm very open to suggestions for this, I hated doing this as much as you hated reading it.